incorrect comparison can cause index OOR

The comparison here 
https://github.com/pion/ion-sfu/blob/6c36fd4b7486ee153d6931efd0ebfdce21c2564f/pkg/buffer/nack.go#L50
is incorrect and can cause a short circuit failure and OOR error.

```
2020-12-29 17:36:19.996] [DEBUG] [64][subscriber.go][func1] => ice connection state: connected
[2020-12-29 17:36:25.905] [DEBUG] [64][publisher.go][func3] => ice connection state: disconnected
panic: runtime error: index out of range [0] with length 0

goroutine 227 [running]:
github.com/pion/ion-sfu/pkg/buffer.(*nackQueue).remove(0xc00066d800, 0xc0005a7b5a)
        /srv/ion-sfu/pkg/buffer/nack.go:50 +0x179
github.com/pion/ion-sfu/pkg/buffer.(*Bucket).addPacket(0xc0002966c0, 0xc000b3e000, 0x3de, 0x2000, 0x7b5a, 0x51679f, 0xa, 0xc0002691e0)
        /srv/ion-sfu/pkg/buffer/bucket.go:37 +0x2d9
github.com/pion/ion-sfu/pkg/buffer.(*Buffer).calc(0xc000264a50, 0xc000b3e000, 0x3de, 0x2000, 0x16554086103e92af)
        /srv/ion-sfu/pkg/buffer/buffer.go:250 +0x105
github.com/pion/ion-sfu/pkg/buffer.(*Buffer).Write(0xc000264a50, 0xc000b3e000, 0x3de, 0x2000, 0x0, 0x0, 0x0)
        /srv/ion-sfu/pkg/buffer/buffer.go:184 +0x11d
github.com/pion/srtp/v2.(*ReadStreamSRTP).write(0xc000645cc0, 0xc000b3e000, 0x3de, 0x2000, 0xc000b3e000, 0x3e8, 0x2000)
        /home/tmiv74_gmail_com/go/pkg/mod/github.com/pion/srtp/v2@v2.0.0-rc.3.0.20201219001609-888371fe3846/stream_srtp.go:63 +0x56
github.com/pion/srtp/v2.(*SessionSRTP).decrypt(0xc00035f4a0, 0xc000b3e000, 0x3e8, 0x2000, 0x3e8, 0x0)
        /home/tmiv74_gmail_com/go/pkg/mod/github.com/pion/srtp/v2@v2.0.0-rc.3.0.20201219001609-888371fe3846/session_srtp.go:160 +0x1a8
github.com/pion/srtp/v2.(*session).start.func1(0xc00035f4a0, 0xc0001c99b0, 0xc66720, 0xc00035f4a0)
        /home/tmiv74_gmail_com/go/pkg/mod/github.com/pion/srtp/v2@v2.0.0-rc.3.0.20201219001609-888371fe3846/session.go:141 +0x12d
created by github.com/pion/srtp/v2.(*session).start
        /home/tmiv74_gmail_com/go/pkg/mod/github.com/pion/srtp/v2@v2.0.0-rc.3.0.20201219001609-888371fe3846/session.go:120 +0x226
```

--- a/pkg/buffer/nack.go
+++ b/pkg/buffer/nack.go
@@ -47,7 +47,7 @@ func (n *nackQueue) remove(sn uint16) {
 	}
 
 	i := sort.Search(len(n.nacks), func(i int) bool { return n.nacks[i].sn >= extSN })
-	if i > len(n.nacks) || n.nacks[i].sn != extSN {
+	if i >= len(n.nacks) || n.nacks[i].sn != extSN {
 		return
 	}
 	copy(n.nacks[i:], n.nacks[i+1:])

